### PR TITLE
Feat: Add nix

### PR DESCRIPTION
**Pull request by Septias**
Add nix to the project. This is only needed for people who use nix on their computer, everyone else is unaffected.
Another option is to ignore all nix related files.

### CODE REVIEW

**Comment by WofWca** [comment]
Where can I learn about what this actually does?

**Comment by Septias** [comment]
maybe this helps: https://nixos.wiki/wiki/Flakes but I did not find any better resource. There is a reddit thread though.
Basically it sets up my environment by installing nodo, lsp etc. of the correct version

**Comment by hpk42** [comment]
seems that linting is failing?

File: flake.nix [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    let
+      supportedSystems = [ "x86_64-linux" "aarch64-linux" "x86_64-darwin" "aarch64-darwin" ];
+      forEachSupportedSystem = f: nixpkgs.lib.genAttrs supportedSystems (system: f {
+        pkgs = import nixpkgs { inherit system; };
+      });
+    in
+    {
+      devShells = forEachSupportedSystem ({ pkgs }: {
+        default = pkgs.mkShell {
+          packages = with pkgs; [ node2nix nodejs nodePackages.pnpm nodePackages.typescript-language-server ];
```
[the remark this answers is not in the document]
Comment (Septias) [reply flake.nix:16]: one can still use pnpm if one prefers, but can remove.

[the remark this answers is not in the document]
Comment (Septias) [reply flake.nix:16]: Node to nix allows one to install all node packages with nix, also not strictly required

File: flake.nix [hunk 2]
```diff
@@ -0,0 +1,20 @@
+{
+  description = "A Nix-flake-based Node.js development environment";
+
+  inputs.nixpkgs.url = "https://flakehub.com/f/NixOS/nixpkgs/0.1.*.tar.gz";
```
[the remark this answers is not in the document]
Comment (Septias) [reply flake.nix:4]: Would rather go with the stable version here, this is the one used by `https://github.com/the-nix-way/dev-templates`

**Comment by Septias** [comment]
@adbenitez @link2xt lets merge this and move on with the next PR

**Pull request merged**